### ISSUE TITLE
Grafana shows wrong value in alerting compared to graphs

### ISSUE THREAD

**Issue by pavel-mutafchiev** [grafana/grafana#10549]
Hello, 
I am running Grafana and Influxdb with telegraf as the collecting agent in a mostly CentOS environment. 

I am having some trouble with the alerting engine firing off alarms with evidently wrong data. 

I am including a couple of screen shots. In the first you can see the alert history where the alert is triggered:
![capture](https://user-images.githubusercontent.com/35542299/35064949-56a5c822-fbd4-11e7-8a48-7500f42e81ab.PNG)
In the second you can see the query over which the alert rule is ran. 
![capture-2](https://user-images.githubusercontent.com/35542299/35065020-8b8ae1e4-fbd4-11e7-80d2-a404d7679e3b.PNG)

As you can see, the values gathered by the query never exceed 1, and in this case they are all 0 for the period in which the evaluation was performed. running the rule manually right after the alert is triggered shows an average of 0.0something which should not trigger the alarm. 

Please let me know if you need any additional details. 

Best Regards.
Pavel

**Comment by pavel-mutafchiev** [comment]
another example, this time showing manual test right after the alert is triggered:
![untitled](https://user-images.githubusercontent.com/35542299/35065380-cd060062-fbd5-11e7-8913-8ad74b4e817d.png)

**Comment by torkelo** [comment]
it is probably related to non_negative_derivative and your group by time. It can be very sensitive and produce inconsistent results. Or some other reason caused influxdb to return 1.15, make sure you have a min interval set (in data source settings) to the rate that you write data to. 

Also you pass 1ms to on negative derivative, that sounds very low, is that really what you want? You can enable alerting log debugging which will print all queries sent to influxdb and the response, might give you a clue why InfluxDB is sometime returning the wrong thing. 

http://docs.grafana.org/alerting/rules/#troubleshooting

Will close this for now, if you think you found a case where Grafana sends the wrong query to infuxdb or is handling the response from InfluxDB in the wrong way to arrive at a value of 1.15 when it should be close to 0 I will reopen.

**Comment by pavel-mutafchiev** [comment]
I came up with the query while looking for a way to implement disk IO utilization as a percentile value and came across this: https://github.com/influxdata/telegraf/issues/1007

I eventually ended up re-uploading all my dashboards adding group by "name" and at the moment I no longer get strange alerts not corresponding to the graph values. If I  run into the same issue I will enable the alerting log as per your advice. Thanks!
